openhands/server/conversation_manager/standalone_conversation_manager.py

```diff
@@ -155,15 +156,33 @@ async def _cleanup_stale(self):
 
     async def get_running_agent_loops(
         self, user_id: str | None = None, filter_to_sids: set[str] | None = None
-    ) -> set[str]:
-        """Get the running session ids. If a user is supplied, then the results are limited to session ids for that user. If a set of filter_to_sids is supplied, then results are limited to these ids of interest."""
-        items: Iterable[tuple[str, Session]] = self._local_agent_loops_by_sid.items()
+    ) -> list[str]:
+        """Get the running session ids in chronological order (oldest first).
+        
+        If a user is supplied, then the results are limited to session ids for that user.
+        If a set of filter_to_sids is supplied, then results are limited to these ids of interest.
+        
+        Returns:
+            A list of session IDs in chronological order (oldest first).
+        """
+        # Get all items and convert to list for sorting
+        items: list[tuple[str, Session]] = list(self._local_agent_loops_by_sid.items())
+        
+        # Filter items if needed
         if filter_to_sids is not None:
-            items = (item for item in items if item[0] in filter_to_sids)
+            items = [item for item in items if item[0] in filter_to_sids]
         if user_id:
-            items = (item for item in items if item[1].user_id == user_id)
-        sids = {sid for sid, _ in items}
-        return sids
+            items = [item for item in items if item[1].user_id == user_id]
+        
+        # Get metadata for each session
+        sids_with_time = {}  # sid -> last_updated_at
+        for sid, _ in items:
+            metadata = await self.conversation_store.get_metadata(sid)
+            if metadata:
```

Comment:
The thinking here I assume is that metadata would never normally be missing, so we should be returning the same set of results as before.